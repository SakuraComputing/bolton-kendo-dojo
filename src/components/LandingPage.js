import React from 'react'

const LandingPage = () => {
  return (
    <div>
        <h1>Bolton Kendo Dojo</h1>
    </div>
  )
}
export default LandingPage;
