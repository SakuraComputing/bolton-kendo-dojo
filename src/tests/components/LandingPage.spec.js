import { shallow } from 'enzyme';
import React from 'react';
import LandingPage from '../../components/LandingPage.js'

describe('landing page component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LandingPage />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should display h1 text', () => {
        const headerText = wrapper.find('h1').text();
        expect(headerText).toEqual('Bolton Kendo Club');
    });
    it('should display paragraph text', () => {
        const headerText = wrapper.find('p').text();
        expect(headerText).toEqual('Welcome to KNMS / Doshinkenyukai');
    });
});