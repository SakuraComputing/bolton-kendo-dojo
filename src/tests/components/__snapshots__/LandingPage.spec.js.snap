// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-layout__box"
      >
        <h1
          className="box-layout__title"
        >
          Bolton Kendo Club
        </h1>
        <p
          className="box-layout__synopsis"
        >
          Welcome to KNMS / Doshinkenyukai
        </p>
        <a
          className="facebook-link"
          href="https://www.facebook.com/groups/896572830375789/"
        >
          <img
            alt="facebook"
            className="facebook-logo"
            src="/img/facebook.png"
          />
          <div
            className="facebook-text"
          >
            Find us on Facebook here
          </div>
        </a>
      </div>,
      "className": "box-layout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="box-layout__title"
          >
            Bolton Kendo Club
          </h1>,
          <p
            className="box-layout__synopsis"
          >
            Welcome to KNMS / Doshinkenyukai
          </p>,
          <a
            className="facebook-link"
            href="https://www.facebook.com/groups/896572830375789/"
          >
            <img
              alt="facebook"
              className="facebook-logo"
              src="/img/facebook.png"
            />
            <div
              className="facebook-text"
            >
              Find us on Facebook here
            </div>
          </a>,
        ],
        "className": "box-layout__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bolton Kendo Club",
            "className": "box-layout__title",
          },
          "ref": null,
          "rendered": "Bolton Kendo Club",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to KNMS / Doshinkenyukai",
            "className": "box-layout__synopsis",
          },
          "ref": null,
          "rendered": "Welcome to KNMS / Doshinkenyukai",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="facebook"
                className="facebook-logo"
                src="/img/facebook.png"
              />,
              <div
                className="facebook-text"
              >
                Find us on Facebook here
              </div>,
            ],
            "className": "facebook-link",
            "href": "https://www.facebook.com/groups/896572830375789/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "facebook",
                "className": "facebook-logo",
                "src": "/img/facebook.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Find us on Facebook here",
                "className": "facebook-text",
              },
              "ref": null,
              "rendered": "Find us on Facebook here",
              "type": "div",
            },
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-layout__box"
        >
          <h1
            className="box-layout__title"
          >
            Bolton Kendo Club
          </h1>
          <p
            className="box-layout__synopsis"
          >
            Welcome to KNMS / Doshinkenyukai
          </p>
          <a
            className="facebook-link"
            href="https://www.facebook.com/groups/896572830375789/"
          >
            <img
              alt="facebook"
              className="facebook-logo"
              src="/img/facebook.png"
            />
            <div
              className="facebook-text"
            >
              Find us on Facebook here
            </div>
          </a>
        </div>,
        "className": "box-layout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="box-layout__title"
            >
              Bolton Kendo Club
            </h1>,
            <p
              className="box-layout__synopsis"
            >
              Welcome to KNMS / Doshinkenyukai
            </p>,
            <a
              className="facebook-link"
              href="https://www.facebook.com/groups/896572830375789/"
            >
              <img
                alt="facebook"
                className="facebook-logo"
                src="/img/facebook.png"
              />
              <div
                className="facebook-text"
              >
                Find us on Facebook here
              </div>
            </a>,
          ],
          "className": "box-layout__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Bolton Kendo Club",
              "className": "box-layout__title",
            },
            "ref": null,
            "rendered": "Bolton Kendo Club",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to KNMS / Doshinkenyukai",
              "className": "box-layout__synopsis",
            },
            "ref": null,
            "rendered": "Welcome to KNMS / Doshinkenyukai",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="facebook"
                  className="facebook-logo"
                  src="/img/facebook.png"
                />,
                <div
                  className="facebook-text"
                >
                  Find us on Facebook here
                </div>,
              ],
              "className": "facebook-link",
              "href": "https://www.facebook.com/groups/896572830375789/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "facebook",
                  "className": "facebook-logo",
                  "src": "/img/facebook.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Find us on Facebook here",
                  "className": "facebook-text",
                },
                "ref": null,
                "rendered": "Find us on Facebook here",
                "type": "div",
              },
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
